src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs

```diff
@@ -46,6 +47,13 @@ internal RuntimeType GetDeclaringTypeInternal()
 
         public override Module Module => GetRuntimeModule();
         public override bool IsCollectible => m_declaringType.IsCollectible;
+
+        public override bool Equals(object? obj) =>
```

Comment:
We end up not having any specialization for MethodInfo. MethodInfo is the most frequently used type among the ones modified in this PR.

I am not sure whether `MetadataUpdater.IsSupported` specialization for RuntimeFieldInfo, RuntimePropertyInfo and RuntimeEventInfo - that are all less important than RuntimeMethodInfo - is worth the trouble.